### PR TITLE
3743 - Broken links - Add API endpoints & activity logs

### PR DESCRIPTION
**Pull request by yaguzmang**

### CODE REVIEW

File: src/meta/api/endpoint/ApiEndPoint.ts [hunk 1]
```diff
@@ -54,6 +54,13 @@ export const ApiEndPoint = {
       one: () => apiPath('cycle-data', 'contacts', 'contact'),
     },
 
+    Link: {
```
Comment (minotogna) [inline src/meta/api/endpoint/ApiEndPoint.ts]: Better to call this Links

File: src/meta/api/endpoint/ApiEndPoint.ts [hunk 2]
```diff
@@ -54,6 +54,13 @@ export const ApiEndPoint = {
...
+      linksCount: () => apiPath('cycle-data', 'links', 'count'),
```
Comment (minotogna) [inline src/meta/api/endpoint/ApiEndPoint.ts]: And this just count

File: src/server/api/cycleData/link/verifyLinks.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+type Request = CycleRequest
+
+export const verifyLinks = async (req: Request, res: Response) => {
+  try {
+    const { assessmentName, cycleName } = req.query
+
+    const { assessment, cycle } = await AssessmentController.getOneWithCycle({ assessmentName, cycleName })
+    const user = Requests.getUser(req)
+
+    await CycleDataController.Link.visitCycleLinks({ assessment, cycle, user })
```
Comment (minotogna) [inline src/server/api/cycleData/link/verifyLinks.ts]: I would call the controller method just verify

File: src/server/controller/cycleData/link/index.ts [hunk 4]
```diff
@@ -1,7 +1,13 @@
+import { LinkRepository } from 'server/repository/assessmentCycle/link'
+
 import { getAllLinksToVisit } from './getAllLinksToVisit'
+import { update } from './update'
 import { visitCycleLinks } from './visitCycleLinks'
```
Comment (minotogna) [inline src/server/controller/cycleData/link/index.ts]: I would call this Links

File: src/server/controller/cycleData/link/visitCycleLinks/worker.ts [hunk 5]
```diff
@@ -17,7 +21,15 @@ const _getLogKey = (job: VisitCycleLinksJob): string => {
...
   try {
-    const { assessment, cycle } = job.data
+    const { assessment, cycle, user } = job.data
+
+    const target = { jobStatus: 'started' }
+    const message = ActivityLogMessage.linksCheckStarted
+    const section = SectionNames.Admin.links
+    const activityLog = { message, section, target, user }
+    const activityLogParams = { activityLog, assessment, cycle }
+    await ActivityLogRepository.insertActivityLog(activityLogParams)
```
Comment (minotogna) [inline src/server/controller/cycleData/link/visitCycleLinks/worker.ts:31]: Why do we need to persist activity log of the job ?

This pr is a bit big 😅

Comment (yaguzmang) [reply src/server/controller/cycleData/link/visitCycleLinks/worker.ts:31]: We would want to keep track when it starts and ends, no? 
😅 I can revert activity log changes and we can discuss it in the next one 👍

Comment (minotogna) [reply src/server/controller/cycleData/link/visitCycleLinks/worker.ts:31]: why do we want to keep track of when it starts and ends? have you checked, as we discussed during the call, what you can get from bullmq directly ?

Comment (yaguzmang) [reply src/server/controller/cycleData/link/visitCycleLinks/worker.ts:31]: Yes, I checked bullmq and I can get the active jobs and know whether or not they are running. That will be coming in the next PR. These activity logs here are unrelated to that, they are just to have a history of when the jobs have run, as we originally discussed.
I also added an activity log for when the user updates a link (when approving it, for example). I can remove it if you think it's unnecessary.

Comment (minotogna) [reply src/server/controller/cycleData/link/visitCycleLinks/worker.ts:31]: Got it 👌agreed, Maybe it's good we have these activity logs for the job.

File: src/meta/assessment/activityLog.ts [hunk 6]
```diff
@@ -8,19 +8,23 @@ export enum ActivityLogMessage {
...
+  contactUpdate = 'contactUpdate',
   descriptionUpdate = 'descriptionUpdate',
   fileCreate = 'fileCreate',
   invitationAccept = 'invitationAccept',
   invitationAdd = 'invitationAdd',
   invitationRemove = 'invitationRemove',
+  linksCheckCompleted = 'linksCheckCompleted',
+  linksCheckFailed = 'linksCheckFailed',
+  linksCheckStarted = 'linksCheckStarted',
+  linkUpdate = 'linkUpdate',
```
Comment (minotogna) [inline src/meta/assessment/activityLog.ts:21]: I wouldn't use for theses logs the past .
E.g. linksCheckStart , etc ... like for linkUpdate and other activities

Comment (yaguzmang) [reply src/meta/assessment/activityLog.ts:21]: Updated it 👍,  thanks

**Review by minotogna — APPROVED** [review]

**Pull request merged**